src/Client.php

```diff
@@ -92,49 +90,44 @@ class Client extends BaseClient
-     *                                   * logger      Inject a Contentful logger
-     *                                   * uriOverride Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
-     *                                   * cache       Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
-     *                                   * autoWarmup  Warm up the cache automatically
+     *                                   * guzzle     Override the guzzle instance used by the Contentful client
+     *                                   * logger     Inject a Contentful logger
+     *                                   * baseUri    Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
+     *                                   * cache      Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
+     *                                   * autoWarmup Warm up the cache automatically
      */
     public function __construct($token, $spaceId, $preview = false, $defaultLocale = null, array $options = [])
     {
-        $baseUri = $preview ? 'https://preview.contentful.com/' : 'https://cdn.contentful.com/';
-
         $options = \array_replace([
             'guzzle' => null,
             'logger' => null,
-            'uriOverride' => null,
+            'baseUri' => null,
             'cacheDir' => null,
             'cache' => null,
             'autoWarmup' => false,
         ], $options);
 
-        $guzzle = $options['guzzle'];
-        $logger = $options['logger'];
-        $uriOverride = $options['uriOverride'];
-        $this->autoWarmup = $options['autoWarmup'];
-
-        if (null !== $uriOverride) {
-            $baseUri = $uriOverride;
+        $baseUri = $preview ? self::URI_PREVIEW : self::URI_DELIVERY;
+        if (null !== $options['baseUri']) {
+            $baseUri = $options['baseUri'];
+            if ('/' === \mb_substr($baseUri, -1)) {
+                $baseUri = \mb_substr($baseUri, 0, -1);
+            }
         }
-        $baseUri .= 'spaces/';
 
-        parent::__construct($token, $baseUri.$spaceId.'/', $logger, $guzzle);
+        parent::__construct($token, $baseUri, $options['logger'], $options['guzzle']);
 
         $this->preview = $preview;
-        $this->instanceCache = new InstanceCache();
         $this->defaultLocale = $defaultLocale;
         $this->spaceId = $spaceId;
 
-        $this->cacheItemPool = $options['cache'] ?: new VoidCachePool();
-
-        if (!$this->cacheItemPool instanceof CacheItemPoolInterface) {
+        $cacheItemPool = $options['cache'] ?: new VoidCachePool();
```

Comment:
Do you do this as a way to turn off caching or is it just to provide a default?

If a default is desired, it should probably not be a VoidCache, but rather an in memory cache.

If having a null cache is desired, then this is OK.

Please clarify.

Reply:
It's a default for not storing anything in cache. It adheres to the interface that `InstanceRepository` requires, except it doesn't retain anything in memory.